src/shogun/mathematics/Math.h

```diff
@@ -1110,9 +1114,11 @@ class CMath : public CSGObject
 		/// Returns a Gaussian or Normal random number.
 		/// Using the polar form of the Box-Muller transform.
 		/// http://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform#Polar_form
-		static inline float64_t normal_random(float64_t mean, float64_t std_dev)
+		static inline float64_t
+		normal_random(float64_t mean, float64_t std_dev, int32_t seed = 12345)
 		{
-			return sg_rand->normal_distrib(mean, std_dev);
+			float64_t result = m_rng->normal_distrib(mean, std_dev);
```

Comment:
no, I somehow forget to remove the seed in here.  If we want to init the seed, we should call CMath::init_random(seed)